src/ex4/server/model/todolist-model.js

```diff
@@ -0,0 +1,95 @@
+import {readData, writeData} from './data-acess.js'
+
+export default class TodoListModel {
+    constructor() {
+        this.todoList = []
+    }
+
+    loadDataFromFile(){
+        
+        try {
+            const data = readData()
+            if(data === null){
+                this.todoList = []
+            }
+            else{
+                this.todoList = JSON.parse(data)
+            } 
+        } catch (err) {
+            console.error(err);
+        }
+    }
+
+    addData(enterValue){
+        this.todoList.push(enterValue)
+    }
+
+    removeData(index){
+        let copyArr = this.todoList
+        copyArr.splice(index,1)
+        this.todoList = copyArr
+    }
+
+    saveDataToFile(){ 
+        writeData(this.todoList)
+    }
+
+    clearAllData(){
+        this.todoList = []
+    }
+
+    orderDataAlphabetically(){
+        this.todoList = TodoListModel.compareFunc(this.todoList,"title")
+    }
+
+    orderDataAlphabeticallyReverse(){
+        this.todoList = TodoListModel.compareFunc(this.todoList,"title").reverse()
+    }
+
+    orderUnDoneToDone(){
+        this.todoList = TodoListModel.compareFunc(this.todoList,"done")
+    }
+
+    orderDoneToUnDone(){
+        this.todoList = TodoListModel.compareFunc(this.todoList,"done").reverse()
```

Comment:
ordering/filtering the todos shouldn't affect the actual data. it should be done dynamically before returning the todos to the user (it will break all the index based logic)